Actions/AL-Go-Helper.ps1

```diff
@@ -1785,11 +1785,21 @@ Function AnalyzeProjectDependencies {
     $projects | ForEach-Object {
         $project = $_
         Write-Host "- $project"
-        $apps = @()
-        $folders = @(Get-ChildItem -Path (Join-Path $baseFolder $project) -Recurse | Where-Object { $_.PSIsContainer -and (Test-Path (Join-Path $_.FullName 'app.json')) } | ForEach-Object { $_.FullName.Substring($baseFolder.Length+1) } )
+
+        # Read project settings
```

Comment:
You can use:

```
$projectSettings = ReadSettings -baseFolder $baseFolder -project $thisProject
$projectSettings = AnalyzeRepo -settings $projectSettings -baseFolder $baseFolder -project $thisProject -doNotCheckArtifactSetting -doNotIssueWarnings
$folders = @($projectSettings.appFolders+$projectSettings.testFolders+$projectSettings.bcptTestFolders)
```

to get projectSettings (and folders) with appFolders and testFolders calculated for you.

Reply:
Thanks. I will try it out.

Reply:
With `AnalyzeRepo` _appFolders_, _testFolders_ and _bcptTestFolders_ are calculated relative to the AL-Go project's folder.

However, `Sort-AppFoldersByDependencies` expects them relative to the root folder (`$baseFolder`).

Reply:
You are right - it was more to have a single place to enumerate and find appFolders etc. (which I apparently didn't honor myself in this example).
You could then instead of setting folders directly use:
```
$folders = @($projectSettings.appFolders+$projectSettings.testFolders+$projectSettings.bcptTestFolders | ForEach-Object { Join-Path $project $_ } )
```
If you decide not to go with this approach, then note that $folders in your code can be null - and $folders.count will fail instead of being 0.

Reply:
Should be alright now.